### DISCUSSION TITLE
handling update for script

### DISCUSSION THREAD

**Discussion by Tealk** [writefreely/writefreely#488] [Questions / Support] [answered]
Hello,

I have recently added the writefreely update to my shell script. Can I always run the migrate command or can this cause problems?

https://codeberg.org/Tealk/shell-administration-script/src/branch/develop/.manage/web/upgrade#L61

**Comment by thebaer** [comment, marked as answer]
Yes, that will work perfectly! Those commands are meant to be run repeatedly without any issues. (You'll notice it in any log messages that come up, as well.)

**Reply by Tealk** [reply]
ok thanks for the answer
a permanent link for the latest version is not currently available?

**Reply by thebaer** [reply]
What do you mean by a "permanent link for the latest version"? Like a single URL that will always point to the latest version?

**Reply by Tealk** [reply]
yes exactly, a link that always point to the downloads of the latest version

**Reply by thebaer** [reply]
This GitHub link should work! https://github.com/writefreely/writefreely/releases/latest

I'll look at adding a permanent link on https://writefreely.org for this, too.

**Reply by Tealk** [reply]
> This GitHub link should work! https://github.com/writefreely/writefreely/releases/latest

but a link to a .tar.gz file would be required, like:
https://github.com/writefreely/writefreely/releases/latest/writefreely_linux_amd64.tar.gz
then i can automate the download without a query for the version number

**Reply by thebaer** [reply]
According to [this page](https://docs.github.com/en/github/administering-a-repository/releasing-projects-on-github/linking-to-releases), it would be: http://github.com/writefreely/writefreely/releases/latest/download/writefreely_0.13.1_linux_amd64.tar.gz

**Reply by Tealk** [reply]
unfortunately there is also the version number in there which changes with every version
so it is not a permanent link

**Reply by thebaer** [reply]
Right, I'll need to set something up on writefreely.org that handles everything correctly. I'll update you when we have that set up.

**Reply by Tealk** [reply]
super thank you
then i can update the script automatically.

**Comment by Tealk** [comment]
I would have another question
with the 0.13 you had to regenerate keys, do you have to delete the keys before or are they replaced when executing the command?

**Reply by thebaer** [reply]
Sorry for the delayed response! No, once you've created the keys, you should never delete them, since it will prevent you from decrypting certain information in the database, and mess up logged in user sessions.

**Reply by Tealk** [reply]
And what is then with the update v0.12.0 there is said in the update description to generate new key

**Reply by thebaer** [reply]
In this latest update, we added a _new_ key for something else, which you can create with that same "generate keys" command. So the command is safe to run at every update, since it'll only create new keys but not automatically delete old ones.
